connection.go

```diff
@@ -341,7 +341,7 @@ func (c *Connection) connect(ctx context.Context) error {
 	// connect
 	transport, err := c.transport.Dial(ctx)
 	if err != nil {
-		c.log.Warning("Connection: error dialing transport: %#v", err)
+		c.log.Debug("Connection: error dialing transport: %#v", err)
```

Comment:
Hmm yeah now that I look again I think the `Dial` error makes sense to be warn. My thinking for the `OnConnect` error is that it is from the client code anyway, so it can take responsibility for printing an error message out beyond debug mode if it wants.